src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -1797,15 +1797,36 @@ private function registerMessengerConfiguration(array $config, ContainerBuilder
             $container->setAlias('messenger.default_serializer', $config['serializer']['default_serializer']);
         }
 
+        $failureTransports = [];
+        if ($config['failure_transport']) {
+            if (!isset($config['transports'][$config['failure_transport']])) {
+                throw new LogicException(sprintf('Invalid Messenger configuration: the failure transport "%s" is not a valid transport or service id.', $config['failure_transport']));
+            }
+
+            $container->setAlias('messenger.failure_transports.default', $config['failure_transport']);
```

Comment:
Wait... shouldn't the target service id here (the 2nd argument) be `'messenger.transport.'.$config['failure_transport']`?

I don't love hard-coding that `'messenger.transport.'.$transportName` id pattern in 2 places (it's main place is below)m but I think we'll need to.